File: elixir_euler/lib/problem_001.ex
# Problem 1: Multiples of 3 and 5
# https://projecteuler.net/problem=1

defmodule Problem001 do
  def sum_of_multiples(x, y, limit) do
    Enum.filter(1..limit, &(rem(&1, x) == 0) || (rem(&1, y) == 0))
      |> Enum.sum
  end
end

File: haskell_euler/problem001.hs
-- Problem 1: Find the sum of all the multiples of 3 or 5 below 1000.
-- https://projecteuler.net/problem=1

-- Using list comprehensions

sumOfMultiples :: Integral a => a -> a
sumOfMultiples (limit) = sum [ x | x <- [1..limit], x `mod` 3 == 0 || x `mod` 2 == 0 ]

sumOfMultiples 1000
-- 334167
